### ISSUE TITLE
`ember deploy` should return non-0 exit code on failure

### ISSUE THREAD

**Issue by v-shan** [ember-cli-deploy/ember-cli-deploy#417]
Hi all,

We have a jenkins build that deploys our assets to S3. The build shows as successful even when the deploy fails because the `ember deploy` command is not returning a non-0 exit code. 

Failure:

```
23:12:06 +- build
23:12:06 |  |
23:12:06 |  +- build
23:12:06 |    - building app to `tmp/deploy-dist` using buildEnv `staging`...
23:12:06 |    - build failed
23:12:06 |
23:12:06 +- didFail
23:12:06 Error: The Broccoli Plugin: [SourceMapConcat] failed with:
23:12:06 Error: ENOENT, no such file or directory '/project/tmp/source_map_concat-input_base_path-efSEw67B.tmp/0/lib/some_file.js'
```

Code snippet:

```
#!/usr/bin/env ruby

def run(cmd)
  puts cmd
  puts `#{cmd}`
  exit($?) unless $?.success?
end

run("ember deploy staging --verbose")
```

**Comment by ghedamat** [comment]
@v-shan would you be open to investigate a PR to fix this?

**Comment by v-shan** [comment]
Hey @ghedamat might have time next month to take a look

**Comment by duizendnegen** [comment]
Solved by PR #443

**Comment by duizendnegen** [comment]
This already returned a non-0 exit code for failure codes, I'd be happy to help investigate why it wasn't working for your deploy pipeline.
